addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -1464,6 +1464,224 @@ QUnit.test('chat window: switch on TAB', async function (assert) {
+        '.o_ChatWindow.o-folded',
+        "all 2 chat windows should be open"
+    );
+
+    await afterNextRender(() =>
+        document.querySelector(`.o_MessagingMenu_toggler`).click()
+    );
+    await afterNextRender(() =>
+        document.querySelector(`
+            .o_MessagingMenu_dropdownMenu
+            .o_NotificationList_preview[data-thread-local-id="${
+                this.env.models['mail.thread'].find(thread =>
+                    thread.id === 1 &&
+                    thread.model === 'mail.channel'
+                ).localId
+            }"]`
+        ).click()
+    );
+
+    let chatWindows = document.querySelectorAll('.o_ChatWindow');
+
+    await afterNextRender(() =>
+        triggerEvent(
+            chatWindows[2].querySelector('.o_ComposerTextInput_textarea'),
+            'keydown',
+            { key: 'Tab' },
+        )
+    );
+
+    assert.strictEqual(
+        chatWindows[1].querySelector('.o_ComposerTextInput_textarea'),
+        document.activeElement,
+        "the chatWindow on the right should have focus"
+    );
+
+    await afterNextRender(() =>
+        triggerEvent(
+            chatWindows[1].querySelector('.o_ComposerTextInput_textarea'),
+            'keydown',
+            { key: 'Tab' },
+        )
+    );
+
+    assert.strictEqual(
+        chatWindows[0].querySelector('.o_ComposerTextInput_textarea'),
+        document.activeElement,
+        "the last chatWindow on the right should have focus"
+    );
+
+    await afterNextRender(() =>
+        triggerEvent(
+            chatWindows[0].querySelector('.o_ComposerTextInput_textarea'),
+            'keydown',
+            { key: 'Tab' },
+        )
+    );
+
+    assert.strictEqual(
+        chatWindows[2].querySelector('.o_ComposerTextInput_textarea'),
+        document.activeElement,
+        "the first chatWindow should have the focus since tab restart the focus at the beginning"
+    );
+});
+
+QUnit.test('chat window: cycle between chat window (RTL)', async function (assert) {
+    assert.expect(5);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: "channel",
+                id: 1,
+                is_pinned: true,
+                name: "channel1",
+            }, {
+                channel_type: "channel",
+                id: 2,
+                is_minimized: true,
+                is_pinned: true,
+                name: "channel2",
+                state: 'open',
+            }, {
+                channel_type: "channel",
+                id: 3,
+                is_minimized: true,
+                is_pinned: true,
+                name: "channel3",
+                state: 'open',
+            }],
+        },
+    });
+    await this.start({
+        env: {
+            browser: {
+                innerWidth: 1920,
+            },
+        },
+    });
+
+    this.env.messaging.locale.update({
+        textDirection: 'rtl'
+    });
```

Comment:
this could trigger a re-render, so would either add some afterNextAnimationFrames (poor choice) or make changes to provide correct direction to env (looks like `web.test_env:makeTestEnvironment()` has a default non-extendable `env._t.database`, which is what is used by `mail.locale` to set text direction, certainly should make it extendable of some sort?)